kong/plugins/key-auth/handler.lua

```diff
@@ -179,9 +237,43 @@ local function do_authentication(conf)
     return nil, { status = 500, message = "An unexpected error occurred" }
   end
 
-  set_consumer(consumer, credential)
+  if conf.validate_signature then
+    -- 4. Verify sha256
+    local secret = credential.secret
+    local now = math.floor(ngx.time())
```

Comment:
`ngx.time` already returns an integer value (as aopposed to `ngx.now`) so no need for the `math.floor` here.

Reply:
Will be in next commit